src/Google.Storage.V1/CopyObjectOptions.cs

```
@@ -0,0 +1,162 @@
+        /// </summary>
+        public long? IfGenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the existing destination object's
+        /// generation does not match the given value.
+        /// </summary>
+        public long? IfGenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the existing destination object's
+        /// meta-generation matches the given value.
+        /// </summary>
+        public long? IfMetagenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the existing destination object's
+        /// meta-generation does not match the given value.
+        /// </summary>
+        public long? IfMetagenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// generation matches the given value.
+        /// </summary>
+        public long? IfSourceGenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// generation does not match the given value.
+        /// </summary>
+        public long? IfSourceGenerationNotMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// meta-generation matches the given value.
+        /// </summary>
+        public long? IfSourceMetagenerationMatch { get; set; }
+
+        /// <summary>
+        /// Precondition for copying: the object is only copied if the source object's
+        /// meta-generation does not match the given value.
+        /// </summary>
+        public long? IfSourceMetagenerationNotMatch { get; set; }
+
+        internal void ModifyRequest(RewriteRequest request)
```

Add a paramName parameter, rather than hardcoding "options" in the throws?